packages/js/src/ui/components/Inbox.tsx

```diff
@@ -10,9 +11,44 @@ type InboxProps = {
   renderBell?: ({ unreadCount }: { unreadCount: number }) => JSX.Element;
 };
 
-export const Inbox = (props: InboxProps) => {
+const InboxContent = () => {
   const [currentScreen, setCurrentScreen] = createSignal<'inbox' | 'settings'>('inbox');
   const { t } = useLocalization();
+
+  const novu = useNovu();
+
+  const [pages, initialLoading, setEl, { end }] = createInfiniteScroll(async (page) => {
+    return (await novu.feeds.fetch({ page })).data;
+  });
+
+  return (
+    <>
+      <Switch>
+        <Match when={currentScreen() === 'inbox'}>
+          <Header updateScreen={setCurrentScreen} />
+          {t('inbox.title')}
+
+          <Show when={!initialLoading()} fallback={<>Loading</>}>
```

Comment:
❓ **question:** the loading state is part of the ticket, are you planning to add it or do it in a separate ticket?

Reply:
Planning to add it!